<!DOCTYPE html>
<html lang="en">

  
  <head>    
    

    
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>Pascal  Berrang | publications</title>
    <meta name="author" content="Pascal  Berrang" />
    <meta name="description" content="Assistant Professor at University of Birmingham
" />
    <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website" />


    
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous" />

    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/css/all.min.css" integrity="sha256-mUZM63G8m73Mcidfrv5E+Y61y7a12O5mW4ezU3bxqW4=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous">
    <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons">

    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/github.css" media="none" id="highlight_theme_light" />

    
    
    <link rel="shortcut icon" href="data:image/svg+xml,<svg xmlns=%22http://www.w3.org/2000/svg%22 viewBox=%220 0 100 100%22><text y=%22.9em%22 font-size=%2290%22>ðŸŽ“</text></svg>">
    
    <link rel="stylesheet" href="/assets/css/main.css">
    <link rel="canonical" href="https://paberr.github.io/publications/">
    
    
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/native.css" media="none" id="highlight_theme_dark" />

    <script src="/assets/js/theme.js"></script>
    <script src="/assets/js/dark_mode.js"></script>
    

  </head>

  
  <body class="fixed-top-nav ">

    
    <header>

      
      <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top">
        <div class="container">
          <a class="navbar-brand title font-weight-lighter" href="https://paberr.github.io//"><span class="font-weight-bold">Pascal</span>   Berrang</a>
          
          <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar top-bar"></span>
            <span class="icon-bar middle-bar"></span>
            <span class="icon-bar bottom-bar"></span>
          </button>

          <div class="collapse navbar-collapse text-right" id="navbarNav">
            <ul class="navbar-nav ml-auto flex-nowrap">

              
              <li class="nav-item ">
                <a class="nav-link" href="/">about</a>
              </li>
              

              
              <li class="nav-item ">
                <a class="nav-link" href="/cv/">cv</a>
              </li>
              <li class="nav-item ">
                <a class="nav-link" href="/hiring/">hiring</a>
              </li>
              <li class="nav-item active">
                <a class="nav-link" href="/publications/">publications<span class="sr-only">(current)</span></a>
              </li>
              <li class="nav-item ">
                <a class="nav-link" href="/teaching/">teaching</a>
              </li>

              
              <div class="toggle-container">
                <a id="light-toggle">
                  <i class="fas fa-moon"></i>
                  <i class="fas fa-sun"></i>
                </a>
              </div>
            </ul>
          </div>
        </div>
      </nav>
    </header>

    
    <div class="container mt-5">
      
        <div class="post">

          <header class="post-header">
            <h1 class="post-title">publications</h1>
            <p class="post-description"></p>
          </header>

          <article>
            
<div class="publications">
  <h2 class="year">2024</h2>
  <ol class="bibliography"><li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">USENIX</abbr></div>

  
  <div id="wu2024quantifying" class="col-sm-8">
    
    
    <div class="title">Quantifying Privacy Risks of Prompts in Visual Prompt Learning</div>
    
    <div class="author">Wu, Yixin,Â Wen, Rui,Â <a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>,Â 
      <em>Berrang, Pascal</em>,Â <a href="https://mhumbert.com" target="_blank" rel="noopener noreferrer">Humbert, Mathias</a>,Â Shen, Yun,Â and <a href="https://yangzhangalmo.github.io" target="_blank" rel="noopener noreferrer">Zhang, Yang</a>
      
    </div>

    
    <div class="periodical">
      <em>In Proceedings of the 33rd USENIX Security Symposium (Security)</em> 2024
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a href="http://arxiv.org/abs/2310.11970" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">arXiv</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
      <a href="/assets/pdf/usenix_security24.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
    </div>

    
    <div class="abstract hidden">
      <p>Large-scale pre-trained models are increasingly adapted to downstream tasks through a new paradigm called prompt learning. In contrast to fine-tuning, prompt learning does not update the pre-trained modelâ€™s parameters. Instead, it only learns an input perturbation, namely prompt, to be added to the downstream task data for predictions. Given the fast development of prompt learning, a well-generalized prompt inevitably becomes a valuable asset as significant effort and proprietary data are used to create it. This naturally raises the question of whether a prompt may leak the proprietary information of its training data. In this paper, we perform the first comprehensive privacy assessment of prompts learned by visual prompt learning through the lens of property inference and membership inference attacks. Our empirical evaluation shows that the prompts are vulnerable to both attacks. We also demonstrate that the adversary can mount a successful property inference attack with limited cost. Moreover, we show that membership inference attacks against prompts can be successful with relaxed adversarial assumptions. We further make some initial investigations on the defenses and observe that our method can mitigate the membership inference attacks with a decent utility-defense trade-off but fails to defend against property inference attacks. We hope our results can shed light on the privacy risks of the popular prompt learning paradigm. To facilitate the research in this direction, we will share our code and models with the community.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">wu2024quantifying</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Quantifying Privacy Risks of Prompts in Visual Prompt Learning}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Wu, Yixin and Wen, Rui and Backes, Michael and Berrang, Pascal and Humbert, Mathias and Shen, Yun and Zhang, Yang}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 33rd USENIX Security Symposium (Security)}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{USENIX Association}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li></ol>

  <h2 class="year">2023</h2>
  <ol class="bibliography">
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">NDSS</abbr></div>

  
  <div id="esiyok2023accountable" class="col-sm-8">
    
    
    <div class="title">Accountable Javascript Code Delivery</div>
    
    <div class="author">Esiyok, Ilkan,Â 
      <em>Berrang, Pascal</em>,Â Cohn-Gordon, Katriel,Â and Kuennemann, Robert
    </div>

    
    <div class="periodical">
      <em>In Proceedings of the 30th Annual Network and Distributed System Security Symposium (NDSS)</em> 2023
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a href="http://arxiv.org/abs/2202.09795" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">arXiv</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
    </div>

    
    <div class="abstract hidden">
      <p>The internet is a major distribution platform for web applications, but there are no effective transparency and audit mechanisms in place for the web. Due to the ephemeral nature of web applications, a client visiting a website has no guarantee that the code it receives today is the same as yesterday, or the same as other visitors receive. Despite advances in web security, it is thus challenging to audit web applications before they are rendered in the browser. We propose Accountable JS, a browser extension and opt in protocol for accountable delivery of active content on a web page. We prototype our protocol, formally model its security properties with the Tamarin Prover, and evaluate its compatibility and performance impact with case studies including WhatsApp Web, AdSense and Nimiq. Accountability is beginning to be deployed at scale, with Metaâ€™s recent announcement of Code Verify available to all 2 billion WhatsApp users, but there has been little formal analysis of such protocols. We formally model Code Verify using the Tamarin Prover and compare its properties to our Accountable JS protocol. We also compare Code Verifyâ€™s and Accountable JS extensionâ€™s performance impacts on WhatsApp Web.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">esiyok2023accountable</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Accountable Javascript Code Delivery}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Esiyok, Ilkan and Berrang, Pascal and Cohn-Gordon, Katriel and Kuennemann, Robert}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 30th Annual Network and Distributed System Security Symposium (NDSS)}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">WWW</abbr></div>

  
  <div id="wang2023zero" class="col-sm-8">
    
    
    <div class="title">On How Zero-Knowledge Proof Blockchain Mixers Improve, and Worsen User Privacy</div>
    
    <div class="author">Wang, Zhipeng,Â Chaliasos, Stefanos,Â Qin, Kaihua,Â Zhou, Liyi,Â Gao, Lifeng,Â 
      <em>Berrang, Pascal</em>,Â Livshits, Ben,Â and Gervais, Arthur
    </div>

    
    <div class="periodical">
      <em>In Proceedings of the ACM Web Conference 2023</em> 2023
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a href="http://arxiv.org/abs/2201.09035" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">arXiv</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
    </div>

    
    <div class="abstract hidden">
      <p>One of the most prominent and widely-used blockchain privacy solutions are zero-knowledge proof (ZKP) mixers operating on top of smart contract-enabled blockchains. ZKP mixers typically advertise their level of privacy through a so-called anonymity set size, similar to k-anonymity, where a user hides among a set of k other users. In reality, however, these anonymity set claims are mostly inaccurate, as we find through empirical measurements of the currently most active ZKP mixers. We propose five heuristics that, in combination, can increase the probability that an adversary links a withdrawer to the correct depositor on average by 51.94% (108.63%) on the most popular Ethereum (ETH) and Binance Smart Chain (BSC) mixer, respectively. Our empirical evidence is hence also the first to suggest a differing privacy-predilection of users on ETH and BSC. We further identify 105 Decentralized Finance (DeFi) attackers leveraging ZKP mixers as the initial funds and to deposit attack revenue (e.g., from phishing scams, hacking centralized exchanges, and blockchain project attacks).<br><br>State-of-the-art mixers are moreover tightly intertwined with the growing DeFi ecosystem by offering "anonymity mining" (AM) incentives, i.e., mixer users receive monetary rewards for mixing coins. However, contrary to the claims of related work, we find that AM does not always contribute to improving the quality of an anonymity set size of a mixer, because AM tends to attract privacy-ignorant users naively reusing addresses.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">wang2023zero</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{On How Zero-Knowledge Proof Blockchain Mixers Improve, and Worsen User Privacy}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Wang, Zhipeng and Chaliasos, Stefanos and Qin, Kaihua and Zhou, Liyi and Gao, Lifeng and Berrang, Pascal and Livshits, Ben and Gervais, Arthur}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the ACM Web Conference 2023}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">ICML</abbr></div>

  
  <div id="yang2023data" class="col-sm-8">
    
    
    <div class="title">Data Poisoning Attacks Against Multimodal Encoders</div>
    
    <div class="author">Yang, Ziqing,Â He, Xinlei,Â Li, Zheng,Â <a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>,Â <a href="https://mhumbert.com" target="_blank" rel="noopener noreferrer">Humbert, Mathias</a>,Â 
      <em>Berrang, Pascal</em>,Â and <a href="https://yangzhangalmo.github.io" target="_blank" rel="noopener noreferrer">Zhang, Yang</a>
      
    </div>

    
    <div class="periodical">
      <em>In 40th International Conference on Machine Learning (ICML)</em> 2023
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a href="http://arxiv.org/abs/2209.15266" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">arXiv</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
    </div>

    
    <div class="abstract hidden">
      <p>Traditional machine learning (ML) models usually rely on large-scale labeled datasets to achieve strong performance. However, such labeled datasets are often challenging and expensive to obtain. Also, the predefined categories limit the modelâ€™s ability to generalize to other visual concepts as additional labeled data is required. On the contrary, the newly emerged multimodal model, which contains both visual and linguistic modalities, learns the concept of images from the raw text. It is a promising way to solve the above problems as it can use easy-to-collect image-text pairs to construct the training dataset and the raw texts contain almost unlimited categories according to their semantics. However, learning from a large-scale unlabeled dataset also exposes the model to the risk of potential poisoning attacks, whereby the adversary aims to perturb the modelâ€™s training dataset to trigger malicious behaviors in it. Previous work mainly focuses on the visual modality. In this paper, we instead focus on answering two questions: (1) Is the linguistic modality also vulnerable to poisoning attacks? and (2) Which modality is most vulnerable? To answer the two questions, we conduct three types of poisoning attacks against CLIP, the most representative multimodal contrastive learning framework. Extensive evaluations on different datasets and model architectures show that all three attacks can perform well on the linguistic modality with only a relatively low poisoning rate and limited epochs. Also, we observe that the poisoning effect differs between different modalities, i.e., with lower MinRank in the visual modality and with higher Hit@K when K is small in the linguistic modality. To mitigate the attacks, we propose both pre-training and post-training defenses. We empirically show that both defenses can significantly reduce the attack performance while preserving the modelâ€™s utility.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">yang2023data</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Yang, Ziqing and He, Xinlei and Li, Zheng and Backes, Michael and Humbert, Mathias and Berrang, Pascal and Zhang, Yang}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{40th International Conference on Machine Learning (ICML)}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Data Poisoning Attacks Against Multimodal Encoders}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2023}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
</ol>

  <h2 class="year">2022</h2>
  <ol class="bibliography">
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">ESORICS</abbr></div>

  
  <div id="backes2022ssle" class="col-sm-8">
    
    
    <div class="title">A framework for constructing Single Secret Leader Election from MPC</div>
    
    <div class="author">
<a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>,Â 
      <em>Berrang, Pascal</em>,Â Hanzlik, Lucjan,Â and Pryvalov, Ivan
    </div>

    
    <div class="periodical">
      <em>In 27th European Symposium on Research in Computer Security (ESORICS) 2022</em> 2022
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
      <a href="/assets/pdf/esorics22.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
    </div>

    
    <div class="abstract hidden">
      <p>The emergence of distributed digital currencies has raised the need for a reliable consensus mechanism. In proof-of-stake cryptocurrencies, the participants periodically choose a closed set of validators, who can vote and append transactions to the blockchain. Each validator can become a leader with the probability proportional to its stake. Keeping the leader private yet unique until it publishes a new block can significantly reduce the attack vector of an adversary and improve the throughput of the network. The problem of Single Secret Leader Election (SSLE) was first formally defined by Boneh et al. in 2020.<br><br>In this work, we propose a novel framework for constructing  SSLE protocols, which relies on secure multi-party computation (MPC) and satisfies the desired security properties. Our framework does not use any shuffle or sort operations and has a computational cost for $N parties as low as O(N) of basic MPC operations per party. We improve the state-of-the-art for SSLE protocols that do not assume a trusted setup. Moreover, our SSLE scheme efficiently handles weighted elections. That is, for a total weight S of N parties, the associated costs are only increased  by a factor of \logS. When the MPC layer is instantiated with techniques based on Shamirâ€™s secret-sharing, our SSLE  has a communication cost of O(N^2) which is spread over O(\logN) rounds, can tolerate up to t&lt;N/2 of faulty nodes without restarting the protocol, and its security relies on DDH in the random oracle model. When the MPC layer is instantiated with more efficient techniques based on garbled circuits, our SSLE requires all parties to participate, up to N-1$ of which can be malicious, and its security is based on the random oracle model.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">backes2022ssle</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Backes, Michael and Berrang, Pascal and Hanzlik, Lucjan and Pryvalov, Ivan}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{27th European Symposium on Research in Computer Security (ESORICS) 2022}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{A framework for constructing Single Secret Leader Election from MPC}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2022}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">arXiv</abbr></div>

  
  <div id="sha2022finetuning" class="col-sm-8">
    
    
    <div class="title">Fine-Tuning Is All You Need to Mitigate Backdoor Attacks</div>
    
    <div class="author">Sha, Zeyang,Â He, Xinlei,Â 
      <em>Berrang, Pascal</em>,Â <a href="https://mhumbert.com" target="_blank" rel="noopener noreferrer">Humbert, Mathias</a>,Â and <a href="https://yangzhangalmo.github.io" target="_blank" rel="noopener noreferrer">Zhang, Yang</a>
      
    </div>

    
    <div class="periodical">
      <em>arXiv preprint arXiv:2212.09067</em> 2022
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a href="http://arxiv.org/abs/2212.09067" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">arXiv</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
    </div>

    
    <div class="abstract hidden">
      <p>Backdoor attacks represent one of the major threats to machine learning models. Various efforts have been made to mitigate backdoors. However, existing defenses have become increasingly complex and often require high computational resources or may also jeopardize modelsâ€™ utility. In this work, we show that fine-tuning, one of the most common and easy-to-adopt machine learning training operations, can effectively remove backdoors from machine learning models while maintaining high model utility. Extensive experiments over three machine learning paradigms show that fine-tuning and our newly proposed super-fine-tuning achieve strong defense performance. Furthermore, we coin a new term, namely backdoor sequela, to measure the changes in model vulnerabilities to other attacks before and after the backdoor has been removed. Empirical evaluation shows that, compared to other defense methods, super-fine-tuning leaves limited backdoor sequela. We hope our results can help machine learning model owners better protect their models from backdoor threats. Also, it calls for the design of more advanced attacks in order to comprehensively assess machine learning modelsâ€™ backdoor vulnerabilities.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">sha2022finetuning</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Sha, Zeyang and He, Xinlei and Berrang, Pascal and Humbert, Mathias and Zhang, Yang}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{arXiv preprint arXiv:2212.09067}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Fine-Tuning Is All You Need to Mitigate Backdoor Attacks}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2022}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
</ol>

  <h2 class="year">2020</h2>
  <ol class="bibliography"><li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">EuroS&amp;P</abbr></div>

  
  <div id="hagestedt2020membership" class="col-sm-8">
    
    
    <div class="title">Membership Inference Against DNA Methylation Databases</div>
    
    <div class="author">Hagestedt, Inken,Â <a href="https://mhumbert.com" target="_blank" rel="noopener noreferrer">Humbert, Mathias</a>,Â 
      <em>Berrang, Pascal</em>,Â Lehmann, Irina,Â Eils, Roland,Â <a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>,Â and <a href="https://yangzhangalmo.github.io" target="_blank" rel="noopener noreferrer">Zhang, Yang</a>
      
    </div>

    
    <div class="periodical">
      <em>In Proceedings of the 2020 IEEE European Symposium on Security and Privacy (EuroS&amp;P)</em> 2020
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
      <a href="/assets/pdf/eurosp20.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
    </div>

    
    <div class="abstract hidden">
      <p>Biomedical data sharing is one of the key elements fostering the advancement of biomedical research but poses severe risks towards the privacy of individuals contributing their data, as already demonstrated for genomic data. In this paper, we study whether and to which extent DNA methylation data, one of the most important epigenetic elements regulating human health, is prone to membership inference attacks, a critical type of attack that reveals an individualâ€™s participation in a given database. We design and evaluate three different attacks exploiting published summary statistics, among which one is based on machine learning and another is exploiting the dependencies between genome and methylation data. Our extensive evaluation on six datasets containing a diverse set of tissues and diseases collected from more than 1,300 individuals in total shows that such membership inference attacks are effective, even when the targetâ€™s methylation profile is not accessible. It further shows that the machine-learning approach outperforms the statistical attacks, and that learned models are transferable across different datasets.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">hagestedt2020membership</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Membership Inference Against DNA Methylation Databases}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Hagestedt, Inken and Humbert, Mathias and Berrang, Pascal and Lehmann, Irina and Eils, Roland and Backes, Michael and Zhang, Yang}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 2020 IEEE European Symposium on Security and Privacy (EuroS\&amp;P)}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2020}</span><span class="p">,</span>
  <span class="na">organization</span> <span class="p">=</span> <span class="s">{IEEE}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li></ol>

  <h2 class="year">2019</h2>
  <ol class="bibliography">
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">NDSS</abbr></div>

  
  <div id="mlleaks" class="col-sm-8">
    
    
    <div class="title">ML-Leaks: Model and Data Independent Membership Inference Attacks and Defenses on Machine Learning Models</div>
    
    <div class="author">Salem, Ahmed,Â <a href="https://yangzhangalmo.github.io" target="_blank" rel="noopener noreferrer">Zhang, Yang</a>,Â <a href="https://mhumbert.com" target="_blank" rel="noopener noreferrer">Humbert, Mathias</a>,Â 
      <em>Berrang, Pascal</em>,Â Fritz, Mario,Â and <a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>
      
    </div>

    
    <div class="periodical">
      <em>In Proceedings of the 26th Annual Network and Distributed System Security Symposium (NDSS)</em> 2019
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
      <a href="/assets/pdf/ndss19_1.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
      <a href="https://github.com/AhmedSalem2/ML-Leaks" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">Code</a>
    </div>

    
    <div class="abstract hidden">
      <p>Machine learning (ML) has become a core component of many real-world applications and training data is a key factor that drives current progress. This huge success has led Internet companies to deploy machine learning as a service (MLaaS). Recently, the first membership inference attack has shown that extraction of information on the training set is possible in such MLaaS settings, which has severe security and privacy implications.<br><br>However, the early demonstrations of the feasibility of such attacks have many assumptions on the adversary, such as using multiple so-called shadow models, knowledge of the target model structure, and having a dataset from the same distribution as the target modelâ€™s training data. We relax all these key assumptions, thereby showing that such attacks are very broadly applicable at low cost and thereby pose a more severe risk than previously thought. We present the most comprehensive study so far on this emerging and developing threat using eight diverse datasets which show the viability of the proposed attacks across domains. In addition, we propose the first effective defense mechanisms against such broader class of membership inference attacks that maintain a high level of utility of the ML model.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">mlleaks</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{ML-Leaks: Model and Data Independent Membership Inference Attacks and Defenses on Machine Learning Models}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Salem, Ahmed and Zhang, Yang and Humbert, Mathias and Berrang, Pascal and Fritz, Mario and Backes, Michael}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2019}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 26th Annual Network and Distributed System Security Symposium (NDSS)}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
<li>
<div class="row">
  <div class="col-sm-2 abbr">
<abbr class="badge">NDSS</abbr><span class="award badge">Best Paper Award</span>
</div>

  
  <div id="mbeacon" class="col-sm-8">
    
    
    <div class="title">MBeacon: Privacy-Preserving Beacons for DNA Methylation Data</div>
    
    <div class="author">Hagestedt, Inken,Â <a href="https://yangzhangalmo.github.io" target="_blank" rel="noopener noreferrer">Zhang, Yang</a>,Â <a href="https://mhumbert.com" target="_blank" rel="noopener noreferrer">Humbert, Mathias</a>,Â 
      <em>Berrang, Pascal</em>,Â Tang, Haixu,Â Wang, XiaoFeng,Â and <a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>
      
    </div>

    
    <div class="periodical">
      <em>In Proceedings of the 26th Annual Network and Distributed System Security Symposium (NDSS)</em> 2019
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
      <a href="/assets/pdf/ndss19_2.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
    </div>

    
    <div class="abstract hidden">
      <p>The advancement of molecular profiling techniques fuels biomedical research with a deluge of data. To facilitate data sharing, the Global Alliance for Genomics and Health established the Beacon system, a search engine designed to help researchers find datasets of interest. While the current Beacon system only supports genomic data, other types of biomedical data, such as DNA methylation, are also essential for advancing our understanding in the field. In this paper, we propose the first Beacon system for DNA methylation data sharing: MBeacon. As the current genomic Beacon is vulnerable to privacy attacks, such as membership inference, and DNA methylation data is highly sensitive, we take a privacy-by-design approach to construct MBeacon.<br><br>First, we demonstrate the privacy threat, by proposing a membership inference attack tailored specifically to unprotected methylation Beacons. Our experimental results show that 100 queries are sufficient to achieve a successful attack with AUC (area under the ROC curve) above 0.9. To remedy this situation, we propose a novel differential privacy mechanism, namely SVT^2, which is the core component of MBeacon. Extensive experiments over multiple datasets show that SVT^2 can successfully mitigate membership privacy risks without significantly harming utility. We further implement a fully functional prototype of MBeacon which we make available to the research community.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">mbeacon</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{MBeacon: Privacy-Preserving Beacons for DNA Methylation Data}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Hagestedt, Inken and Zhang, Yang and Humbert, Mathias and Berrang, Pascal and Tang, Haixu and Wang, XiaoFeng and Backes, Michael}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2019}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 26th Annual Network and Distributed System Security Symposium (NDSS)}</span><span class="p">,</span>
  <span class="na">award</span> <span class="p">=</span> <span class="s">{Best Paper Award}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">CVCBT</abbr></div>

  
  <div id="berrang2019albatross" class="col-sm-8">
    
    
    <div class="title">Albatross â€“ An optimistic consensus algorithm</div>
    
    <div class="author">
      <em>Berrang, Pascal</em>,Â Styp-Rekowsky, Philipp,Â Wissfeld, Marvin,Â FranÃ§a, Bruno,Â and Trinkler, Reto
    </div>

    
    <div class="periodical">
      <em>In Proceedings of the Crypto Valley Conference on Blockchain Technology (CVCBT)</em> 2019
    </div>

    
    <div class="links">
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
    </div>

    
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">berrang2019albatross</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Albatross -- An optimistic consensus algorithm}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Berrang, Pascal and von Styp-Rekowsky, Philipp and Wissfeld, Marvin and Fran{\c{c}}a, Bruno and Trinkler, Reto}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the Crypto Valley Conference on Blockchain Technology (CVCBT)}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2019}</span><span class="p">,</span>
  <span class="na">organization</span> <span class="p">=</span> <span class="s">{IEEE}</span><span class="p">,</span>
  <span class="na">shorthand</span> <span class="p">=</span> <span class="s">{CryptoValley19}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">PoPETs</abbr></div>

  
  <div id="sahlem2019privacy" class="col-sm-8">
    
    
    <div class="title">Privacy-Preserving Similar Patient Queries for Combined Biomedical Data</div>
    
    <div class="author">Salem, Ahmed,Â 
      <em>Berrang, Pascal</em>,Â <a href="https://mhumbert.com" target="_blank" rel="noopener noreferrer">Humbert, Mathias</a>,Â and <a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>
      
    </div>

    
    <div class="periodical">
      <em>Proceedings on Privacy Enhancing Technologies (PoPETs)</em> 2019
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
      <a href="/assets/pdf/popets19.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
    </div>

    
    <div class="abstract hidden">
      <p>The decreasing costs of molecular profiling have fueled the biomedical research community with a plethora of new types of biomedical data, enabling a breakthrough towards more precise and personalized medicine. Naturally, the increasing availability of data also enables physicians to compare patientsâ€™ data and treatments easily and to find similar patients in order to propose the optimal therapy. Such similar patient queries (SPQs) are of utmost importance to medical practice and will be relied upon in future health information exchange systems. While privacy-preserving solutions have been previously studied, those are limited to genomic data, ignoring the different newly available types of biomedical data.<br><br>In this paper, we propose new cryptographic techniques for finding similar patients in a privacy-preserving manner with various types of biomedical data, including genomic, epigenomic and transcriptomic data as well as their combination. We design protocols for two of the most common similarity metrics in biomedicine: the Euclidean distance and Pearson correlation coefficient. Moreover, unlike previous approaches, we account for the fact that certain locations contribute differently to a given disease or phenotype by allowing to limit the query to the relevant locations and to assign them different weights. Our protocols are specifically designed to be highly efficient in terms of communication and bandwidth, requiring only one or two rounds of communication and thus enabling scalable parallel queries. We rigorously prove our protocols to be secure based on cryptographic games and instantiate our technique with three of the most important types of biomedical data â€“ namely DNA, microRNA expression, and DNA methylation. Our experimental results show that our protocols can compute a similarity query over a typical number of positions against a database of 1,000 patients in a few seconds. Finally, we propose and formalize strategies to mitigate the threat of malicious users or hospitals.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">sahlem2019privacy</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Salem, Ahmed and Berrang, Pascal and Humbert, Mathias and Backes, Michael}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Privacy-Preserving Similar Patient Queries for Combined Biomedical Data}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Proceedings on Privacy Enhancing Technologies (PoPETs)}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{2019}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{1}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2019}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
</ol>

  <h2 class="year">2018</h2>
  <ol class="bibliography">
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">PiMLAI</abbr></div>

  
  <div id="revisiting" class="col-sm-8">
    
    
    <div class="title">Revisiting Membership Inference Attacks Against Machine Learning Models</div>
    
    <div class="author">Salem, Ahmed,Â <a href="https://yangzhangalmo.github.io" target="_blank" rel="noopener noreferrer">Zhang, Yang</a>,Â <a href="https://mhumbert.com" target="_blank" rel="noopener noreferrer">Humbert, Mathias</a>,Â 
      <em>Berrang, Pascal</em>,Â Fritz, Mario,Â and <a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>
      
    </div>

    
    <div class="periodical">
      <em>In Privacy in Machine Learning and Artificial Intelligence (PiMLAI)</em> 2018
    </div>

    
    <div class="links">
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
    </div>

    
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">revisiting</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Revisiting Membership Inference Attacks Against Machine Learning Models}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Salem, Ahmed and Zhang, Yang and Humbert, Mathias and Berrang, Pascal and Fritz, Mario and Backes, Michael}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2018}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Privacy in Machine Learning and Artificial Intelligence (PiMLAI)}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">EuroS&amp;P</abbr></div>

  
  <div id="backes2017dissecting" class="col-sm-8">
    
    
    <div class="title">Dissecting Privacy Risks in Biomedical Data</div>
    
    <div class="author">
      <em>Berrang, Pascal</em>,Â <a href="https://mhumbert.com" target="_blank" rel="noopener noreferrer">Humbert, Mathias</a>,Â <a href="https://yangzhangalmo.github.io" target="_blank" rel="noopener noreferrer">Zhang, Yang</a>,Â Lehmann, Irina,Â Eils, Roland,Â and <a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>
      
    </div>

    
    <div class="periodical">
      <em>In Proceedings of the 2018 IEEE European Symposium on Security and Privacy (EuroS&amp;P)</em> 2018
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
      <a href="/assets/pdf/eurosp18.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
    </div>

    
    <div class="abstract hidden">
      <p>The decreasing costs of molecular profiling has fueled the biomedical research community with a plethora of new types of biomedical data, enabling a breakthrough towards a more precise and personalized medicine. However, the release of these intrinsically highly sensitive data poses a new severe privacy threat. While biomedical data is largely associated with our health, there also exist various correlations between different types of biomedical data, along the temporal dimension, and also in-between family members. However, so far, the security community has focused on privacy risks stemming from genomic data, largely overlooking the manifold interdependencies between other biomedical data.<br><br>In this paper, we present a generic framework for quantifying the privacy risks in biomedical data taking into account the various interdependencies between data (i) of different types, (ii) from different individuals, and (iii) at different time. To this end, we rely on a Bayesian network model that allows us to take all aforementioned dependencies into account and run exact probabilistic inference attacks very efficiently. Furthermore, we introduce a generic algorithm for building the Bayesian network, which encompasses expert knowledge for known dependencies, such as genetic inheritance laws, and learns previously unknown dependencies from the data. Then, we conduct a thorough inference risk evaluation with a very rich dataset containing genomic and epigenomic data of mothers and children over multiple years. Besides effective probabilistic inference, we further demonstrate that our Bayesian network model can also serve as a building block for other attacks. We show that, with our framework, an adversary can efficiently identify the parent-child relationships based on methylation data with a success rate of 95%.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">backes2017dissecting</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Dissecting Privacy Risks in Biomedical Data}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Berrang, Pascal and Humbert, Mathias and Zhang, Yang and Lehmann, Irina and Eils, Roland and Backes, Michael}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 2018 IEEE European Symposium on Security and Privacy (EuroS\&amp;P)}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{IEEE}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2018}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
</ol>

  <h2 class="year">2017</h2>
  <ol class="bibliography"><li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">S&amp;P</abbr></div>

  
  <div id="backes2017meQTLs" class="col-sm-8">
    
    
    <div class="title">Identifying Personal DNA Methylation Profiles by Genotype Inference</div>
    
    <div class="author">
<a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>,Â 
      <em>Berrang, Pascal</em>,Â Bieg, Matthias,Â Eils, Roland,Â Herrmann, Carl,Â <a href="https://mhumbert.com" target="_blank" rel="noopener noreferrer">Humbert, Mathias</a>,Â and Lehmann, Irina
    </div>

    
    <div class="periodical">
      <em>In Proceedings of the 38th IEEE Symposium on Security and Privacy (S&amp;P)</em> 2017
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
      <a href="/assets/pdf/oakland17.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
      <a href="https://github.com/paberr/ciphermed-forests" class="btn btn-sm z-depth-0" role="button" target="_blank" rel="noopener noreferrer">Code</a>
    </div>

    
    <div class="abstract hidden">
      <p>Since the first whole-genome sequencing, the biomedical research community has made significant steps towards a more precise, predictive and personalized medicine. Genomic data is nowadays widely considered privacy-sensitive and consequently protected by strict regulations and released only after careful consideration. Various additional types of biomedical data, however, are not shielded by any dedicated legal means and consequently disseminated much less thoughtfully. This in particular holds true for DNA methylation data as one of the most important and well-understood epigenetic element influencing human health.<br><br>In this paper, we show that, in contrast to the aforementioned belief, releasing oneâ€™s DNA methylation data causes privacy issues akin to releasing oneâ€™s actual genome. We show that already a small subset of methylation regions influenced by genomic variants are sufficient to infer parts of someoneâ€™s genome, and to further map this DNA methylation profile to the corresponding genome. Notably, we show that such re-identification is possible with 97.5% accuracy, relying on a dataset of more than 2500 genomes, and that we can reject all wrongly matched genomes using an appropriate statistical test. We provide means for countering this threat by proposing a novel cryptographic scheme for privately classifying tumors that enables a privacy-respecting medical diagnosis in a common clinical setting. The scheme relies on a combination of random forests and homomorphic encryption, and it is proven secure in the honest-but-curious model. We evaluate this scheme on real DNA methylation data, and show that we can keep the computational overhead to acceptable values for our application scenario.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">backes2017meQTLs</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Identifying Personal {DNA} Methylation Profiles by Genotype Inference}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Backes, Michael and Berrang, Pascal and Bieg, Matthias and Eils, Roland and Herrmann, Carl and Humbert, Mathias and Lehmann, Irina}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 38th IEEE Symposium on Security and Privacy (S\&amp;P)}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2017}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{IEEE}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li></ol>

  <h2 class="year">2016</h2>
  <ol class="bibliography">
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">USENIX</abbr></div>

  
  <div id="backes2016epigenetic_privacy" class="col-sm-8">
    
    
    <div class="title">Privacy in Epigenetics: Temporal Linkability of MicroRNA Expression Profiles</div>
    
    <div class="author">
<a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>,Â 
      <em>Berrang, Pascal</em>,Â Hecksteden, Anne,Â <a href="https://mhumbert.com" target="_blank" rel="noopener noreferrer">Humbert, Mathias</a>,Â Keller, Andreas,Â and Meyer, Tim
    </div>

    
    <div class="periodical">
      <em>In Proceedings of the 25th USENIX Security Symposium (Security)</em> 2016
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
      <a href="/assets/pdf/usenix_security16.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
    </div>

    
    <div class="abstract hidden">
      <p>The decreasing cost of molecular profiling tests, such as DNA sequencing, and the consequent increasing availability of biological data are revolutionizing medicine, but at the same time create novel privacy risks. The research community has already proposed a plethora of methods for protecting genomic data against these risks. However, the privacy risks stemming from epigenetics, which bridges the gap between the genome and our health characteristics, have been largely overlooked so far, even though epigenetic data such as microRNAs (miRNAs) are no less privacy sensitive. This lack of investigation is attributed to the common belief that the inherent temporal variability of miRNAs shields them from being tracked and linked over time.<br><br>In this paper, we show that, contrary to this belief, miRNA expression profiles can be successfully tracked over time, despite their variability. Specifically, we show that two blood-based miRNA expression profiles taken with a time difference of one week from the same person can be matched with a success rate of 90%. We furthermore observe that this success rate stays almost constant when the time difference is increased from one week to one year. In order to mitigate the linkability threat, we propose and thoroughly evaluate two countermeasures: (i) hiding a subset of disease-irrelevant miRNA expressions, and (ii) probabilistically sanitizing the miRNA expression profiles. Our experiments show that the second mechanism provides a better trade-off between privacy and disease-prediction accuracy.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">backes2016epigenetic_privacy</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Privacy in Epigenetics: Temporal Linkability of Micro{RNA} Expression Profiles}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Backes, Michael and Berrang, Pascal and Hecksteden, Anne and Humbert, Mathias and Keller, Andreas and Meyer, Tim}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 25th USENIX Security Symposium (Security)}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2016}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{USENIX Association}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">CCS</abbr></div>

  
  <div id="backes2016membership" class="col-sm-8">
    
    
    <div class="title">Membership Privacy in MicroRNA-based Studies</div>
    
    <div class="author">
<a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>,Â 
      <em>Berrang, Pascal</em>,Â <a href="https://mhumbert.com" target="_blank" rel="noopener noreferrer">Humbert, Mathias</a>,Â and Manoharan, Praveen
    </div>

    
    <div class="periodical">
      <em>In Proceedings of the 23rd ACM Conference on Computer and Communication Security (CCS)</em> 2016
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
      <a href="/assets/pdf/ccs16.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
    </div>

    
    <div class="abstract hidden">
      <p>The continuous decrease in cost of molecular profiling tests is revolutionizing medical research and practice, but it also raises new privacy concerns. One of the first attacks against privacy of biological data, proposed by Homer et al. in 2008, showed that, by knowing parts of the genome of a given individual and summary statistics of a genome-based study, it is possible to detect if this individual participated in the study. Since then, a lot of work has been carried out to further study the theoretical limits and to counter the genome-based membership inference attack. However, genomic data are by no means the only or the most influential biological data threatening personal privacy. For instance, whereas the genome informs us about the risk of developing some diseases in the future, epigenetic biomarkers, such as microRNAs, are directly and deterministically affected by our health condition including most common severe diseases.<br><br>In this paper, we show that the membership inference attack also threatens the privacy of individuals contributing their microRNA expressions to scientific studies. Our results on real and public microRNA expression data demonstrate that disease-specific datasets are especially prone to membership detection, offering a true-positive rate of up to 77% at a false-negative rate of less than 1%. We present two attacks: one relying on the L1 distance and the other based on the likelihood-ratio test. We show that the likelihood-ratio test provides the highest adversarial success and we derive a theoretical limit on this success. In order to mitigate the membership inference, we propose and evaluate both a differentially private mechanism and a hiding mechanism. We also consider two types of adversarial prior knowledge for the differentially private mechanism and show that, for relatively large datasets, this mechanism can protect the privacy of participants in miRNA-based studies against strong adversaries without degrading the data utility too much. Based on our findings and given the current number of miRNAs, we recommend to only release summary statistics of datasets containing at least a couple of hundred individuals.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">backes2016membership</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Membership Privacy in Micro{RNA}-based Studies}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Backes, Michael and Berrang, Pascal and Humbert, Mathias and Manoharan, Praveen}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 23rd ACM Conference on Computer and Communication Security (CCS)}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2016}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{ACM}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">UEOP</abbr></div>

  
  <div id="backes2016epigenomic" class="col-sm-8">
    
    
    <div class="title">On Epigenomic Privacy: Tracking Personal MicroRNA Expression Profiles over Time</div>
    
    <div class="author">
<a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>,Â 
      <em>Berrang, Pascal</em>,Â Hecksteden, Anne,Â <a href="https://mhumbert.com" target="_blank" rel="noopener noreferrer">Humbert, Mathias</a>,Â Keller, Andreas,Â and Meyer, Tim
    </div>

    
    <div class="periodical">
      <em>In Workshop on Understanding and Enhancing Online Privacy (UEOP), affiliated with NDSS</em> 2016
    </div>

    
    <div class="links">
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
    </div>

    
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">backes2016epigenomic</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{On Epigenomic Privacy: Tracking Personal Micro{RNA} Expression Profiles over Time}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Backes, Michael and Berrang, Pascal and Hecksteden, Anne and Humbert, Mathias and Keller, Andreas and Meyer, Tim}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Workshop on Understanding and Enhancing Online Privacy (UEOP), affiliated with NDSS}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2016}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">GenoPri</abbr></div>

  
  <div id="backes2016erosion" class="col-sm-8">
    
    
    <div class="title">Simulating the Large-scale Erosion of Genomic Privacy Over Time</div>
    
    <div class="author">
<a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>,Â 
      <em>Berrang, Pascal</em>,Â <a href="https://mhumbert.com" target="_blank" rel="noopener noreferrer">Humbert, Mathias</a>,Â Shen, Xiaoyu,Â and Wolf, Verena
    </div>

    
    <div class="periodical">
      <em>In 3rd International Workshop on Genome Privacy and Security (GenoPri), Selected for publication in IEEE/ACM Transactions on Computational Biology and Bioinformatics</em> 2016
    </div>

    
    <div class="links">
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
      <a href="/assets/pdf/genopri16.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a>
    </div>

    
    <div class="abstract hidden">
      <p>The dramatically decreasing costs of DNA sequencing have triggered more than a million humans to date to have their genotypes sequenced. Moreover, these individuals increasingly make their genomic data publicly available, and thereby create unique privacy threats not only for themselves, but also for their relatives because of their DNA similarities. More generally, an entity that gains access to a significant fraction of sequenced genotypes from a given population might be able to infer even the genomes of unsequenced individuals by relying on available data.<br><br>In this paper, we propose a simulation-based model for quantifying the impact of continuously sequencing and publicizing personal genomic data on a populationâ€™s genomic privacy. Our simulation probabilistically models data sharing by individuals and additionally takes into account the influence on genomic privacy of geopolitical events such as migration, and sociological trends such as interracial marriage. We exemplarily instantiate our simulation with a sample population of 1,000 individuals, and evaluate the evolution of privacy under different settings over either thousands of genomic variants or a subset of variants influencing the phenotype. Our findings notably demonstrate that an increasing sharing rate of genomic data in the future entails a substantial negative effect on the privacy of all older generations. Moreover, we find that mixed populations, due to their large genomic diversity, face a less severe erosion of genomic privacy over time than more homogeneous populations. However, even when no data is shared, the genomic privacy averaged over a large number of variants is already very low since mere population allele frequencies already reveal a lot of information about the values of the genomic variants. By focusing on a subset of sensitive variants, we observe a higher genetic diversity in the population. Thus, genomic-data sharing can be much more detrimental for the privacy of the most sensitive variants.</p>
    </div>
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">backes2016erosion</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Simulating the Large-scale Erosion of Genomic Privacy Over Time}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Backes, Michael and Berrang, Pascal and Humbert, Mathias and Shen, Xiaoyu and Wolf, Verena}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{3rd International Workshop on Genome Privacy and Security (GenoPri), Selected for publication in IEEE/ACM Transactions on Computational Biology and Bioinformatics}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2016}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
<li>
<div class="row">
  <div class="col-sm-2 abbr"></div>

  
  <div id="backes2016zoos" class="col-sm-8">
    
    
    <div class="title">From Zoos to Safaris â€“ From Closed-World Enforcement to Open-World Assessment of Privacy</div>
    
    <div class="author">
<a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>,Â 
      <em>Berrang, Pascal</em>,Â and Manoharan, Praveen
    </div>

    
    <div class="periodical">
      <em>In 3rd International Workshop on Genome Privacy and Security (GenoPri), Selected for publication in IEEE/ACM Transactions on Computational Biology and Bioinformatics</em> 2016
    </div>

    
    <div class="links">
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
    </div>

    
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@incollection</span><span class="p">{</span><span class="nl">backes2016zoos</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{From Zoos to Safaris -- From Closed-World Enforcement to Open-World Assessment of Privacy}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Backes, Michael and Berrang, Pascal and Manoharan, Praveen}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Foundations of Security Analysis and Design VIII}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2016}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Springer-Verlag}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
<li>
<div class="row">
  <div class="col-sm-2 abbr"><abbr class="badge">WPES</abbr></div>

  
  <div id="backes2016profile" class="col-sm-8">
    
    
    <div class="title">Profile Linkability despite Anonymity in Social Media Systems</div>
    
    <div class="author">
<a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>,Â 
      <em>Berrang, Pascal</em>,Â Goga, Oana,Â Gummadi, Krishna,Â and Manoharan, Praveen
    </div>

    
    <div class="periodical">
      <em>In Proceedings of the 15th ACM Workshop on Privacy in the Electronic Society (WPES)</em> 2016
    </div>

    
    <div class="links">
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
    </div>

    
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">backes2016profile</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Profile Linkability despite Anonymity in Social Media Systems}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Backes, Michael and Berrang, Pascal and Goga, Oana and Gummadi, Krishna and Manoharan, Praveen}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2016}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the 15th ACM Workshop on Privacy in the Electronic Society (WPES)}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{ACM}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
</ol>

  <h2 class="year">2015</h2>
  <ol class="bibliography"><li>
<div class="row">
  <div class="col-sm-2 abbr"></div>

  
  <div id="backes2015well" class="col-sm-8">
    
    
    <div class="title">How well do you blend into the crowd?</div>
    
    <div class="author">
<a href="https://cispa.de/en/people/backes" target="_blank" rel="noopener noreferrer">Backes, Michael</a>,Â 
      <em>Berrang, Pascal</em>,Â and Manoharan, Praveen
    </div>

    
    <div class="periodical">
      <em>d-convergence: assessing identity disclosure risks in large-open scale web settings</em> 2015
    </div>

    
    <div class="links">
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
    </div>

    
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">backes2015well</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{How well do you blend into the crowd?}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Backes, Michael and Berrang, Pascal and Manoharan, Praveen}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{d-convergence: assessing identity disclosure risks in large-open scale web settings}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2015}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li></ol>

  <h2 class="year">2012</h2>
  <ol class="bibliography">
<li>
<div class="row">
  <div class="col-sm-2 abbr"></div>

  
  <div id="hartmanns2012comparative" class="col-sm-8">
    
    
    <div class="title">A comparative analysis of decentralized power grid stabilization strategies</div>
    
    <div class="author">Hartmanns, Arnd,Â Hermanns, Holger,Â and <em>Berrang, Pascal</em>
      
    </div>

    
    <div class="periodical">
      <em>In Proceedings of the Winter Simulation Conference</em> 2012
    </div>

    
    <div class="links">
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
    </div>

    
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">hartmanns2012comparative</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{A comparative analysis of decentralized power grid stabilization strategies}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Hartmanns, Arnd and Hermanns, Holger and Berrang, Pascal}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Proceedings of the Winter Simulation Conference}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2012}</span><span class="p">,</span>
  <span class="na">organization</span> <span class="p">=</span> <span class="s">{Winter Simulation Conference}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
<li>
<div class="row">
  <div class="col-sm-2 abbr"></div>

  
  <div id="berrang2012dependability" class="col-sm-8">
    
    
    <div class="title">Dependability results for power grids with decentralized stabilization strategies</div>
    
    <div class="author">
      <em>Berrang, Pascal</em>,Â Bogdoll, Jonathan,Â Moritz, Ernst Hahn,Â Hartmanns, Arnd,Â and Hermanns, Holger
    </div>

    
    <div class="periodical">
      <em>AVACS Technical Report</em> 2012
    </div>

    
    <div class="links">
      <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a>
    </div>

    
    <div class="bibtex hidden">
      
      <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">berrang2012dependability</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Dependability results for power grids with decentralized stabilization strategies}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Berrang, Pascal and Bogdoll, Jonathan and Moritz, Ernst Hahn and Hartmanns, Arnd and Hermanns, Holger}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{AVACS Technical Report}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{83}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2012}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure>
    </div>
  </div>
</div>
</li>
</ol>


</div>

          </article>

        </div>

    </div>

        
    <footer class="fixed-bottom">
      <div class="container mt-0">
        Â© Copyright 2023 Pascal  Berrang. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="noopener noreferrer">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" target="_blank" rel="noopener noreferrer">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="noopener noreferrer">GitHub Pages</a>.
Last updated: October 25, 2023.
      </div>
    </footer>

    
    
  <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>

    
  <script src="https://cdn.jsdelivr.net/npm/@popperjs/core@2.11.2/dist/umd/popper.min.js" integrity="sha256-l/1pMF/+J4TThfgARS6KwWrk/egwuVvhRzfLAMQ6Ds4=" crossorigin="anonymous"></script>
  <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.min.js" integrity="sha256-SyTu6CwrfOhaznYZPoolVw2rxoY7lKYKQvqbtqN93HI=" crossorigin="anonymous"></script>
  <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script>

    
  <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script>
  <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script>
  <script defer src="/assets/js/masonry.js" type="text/javascript"></script>
    
  
  <script src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.6/dist/medium-zoom.min.js" integrity="sha256-EdPgYcPk/IIrw7FYeuJQexva49pVRZNmt3LculEr7zM=" crossorigin="anonymous"></script>
  <script src="/assets/js/zoom.js"></script>
  <script src="/assets/js/common.js"></script>

    
  <script type="text/javascript">
    window.MathJax = {
      tex: {
        tags: 'ams'
      }
    };
  </script>
  <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script>
  <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>

    
  </body>
</html>

